File: 5ku/FirstNonRepeatingCharacter/FirstNonRepeatingCharacter.java
import java.util.HashMap;
import java.util.Map;

public class Kata {
    public static String firstNonRepeatingLetter(String s){
      // Add your code here
      String res = s.toLowerCase();
        Map<String,Integer> m = new HashMap<>();
        for (int i = 0; i < res.length(); i++) {
            if (m.get(String.valueOf(res.charAt(i))) == null) {
                m.put(String.valueOf(res.charAt(i)), 1);
            }else{
                m.put(String.valueOf(res.charAt(i)), m.get(String.valueOf(res.charAt(i))) + 1);
            }
        }
        for (char c : s.toCharArray()) {
          if (m.get(String.valueOf(c).toLowerCase()) != null && m.get(String.valueOf(c).toLowerCase()) == 1) {
            return String.valueOf(c);
          }
        }
        return "";
    }
  }


/*	best
public class Kata {
  public static String firstNonRepeatingLetter(String s){
    for (String str : s.split("")) {
      if (s.toUpperCase().indexOf(str.toUpperCase()) == s.toUpperCase().lastIndexOf(str.toUpperCase())) return str;
     }
    return "";
  }
}
*/

File: 5ku/FirstNonRepeatingCharacter/FirstNonRepeatingCharacter.js
function firstNonRepeatingLetter(s) {
    small = s.toLocaleLowerCase()
    for (const key in s) {
        var regexWithFlags = new RegExp(s[key].toLocaleLowerCase(), 'gi');
        if (small.match(regexWithFlags).length==1) {
            return s[key];
        }
    }
    return ""
    // Add your code here
  }
  
  
/*	best
function firstNonRepeatingLetter(s) {
  for(var i in s) {
    if(s.match(new RegExp(s[i],"gi")).length === 1) {
      return s[i];
    }
  }
  return '';
}
*/
